lib/spack/spack/compiler.py

```diff
@@ -592,7 +595,16 @@ def search_regexps(cls, language):
         # defined for the compiler
         compiler_names = getattr(cls, "{0}_names".format(language))
         prefixes = [""] + cls.prefixes
-        suffixes = [""] + cls.suffixes
+        suffixes = [""]
+        # Windows compilers generally have an extension of some sort
+        # as do most files on Windows, handle that case here
+        if is_windows:
+            ext = r"\.[a-zA-Z]+"
```

Comment:
Would the extension generally be "exe'? I think it would be good to limit this if possible.

Presumably this is because you expect for example `gcc` to be called `gcc.exe` on Windows - did I get that right?

Reply:
I wanted to leave things open ended as a compiler may have a .bat file or something as it's driver script. I can't imagine it's common (and I've never seen it although I only used MSVC and LLVM-clang on Windows) but I didn't want to preclude detection because of a certain extension. It also allows for the tests to create .bat scripts much the same way they mock compilers with extensionless scripts on Unix. I could have constrained this to just .bat and .exe, but as I said, I like to err on the side of not accidently leaving things out. If you disagree I have no problem with that, and I'll happily change things.

Reply:
I think allowing *any* extension may be pulling in too many things (e.g. `.png`), but we could test for `(exe|bat)` for example to cover those). Technically the discovery logic accounts for failing to run something that isn't runnable, but I still think it's worth not trying to run things when it wouldn't make sense.